### ISSUE TITLE
Document/warn/fix improper basenames

### ISSUE THREAD

**Issue by pshrmn** [ReactTraining/history#430]
From what I can tell, a `basename` should never have a trailing slash.
```js
basename = '' // default
basename = '/basename' // yes
basename = '/basename/' // no
```
Currently, a user can pass a `basename` with a trailing slash and the only sign of this being wrong would be that the paths output by history will have a double slash.
```js
"/basename/" + "/link/path" = "/basename//link/path"
```

I can see three ways to handle this:
1. Document that the `basename` must not have a trailing slash.
2. Use `warning` to log an error when the `basename` has a trailing slash.
3. Strip the trailing slash from the `basename` for the user.

I can put together a PR, but I am not sure which approach would be preferred.

**Comment by mjackson** [comment]
We should probably just always assume that the basename should be a "URL join" whenever we use it. I was trying to avoid that and have it be simpler–literally just a string that you put at the beginning of the URL. But people always use it as if it were a URL segment.

**Comment by pshrmn** [comment]
Semi-related: have you thought about removing basenames from hash history? I have a hard time coming up with a reason for them.

**Comment by mjackson** [comment]
Ya, I have. Only use case I can think of is if you were trying to stitch together a bunch of mini sites that all used hash history, so you had to share the hash with other sites. Granted, that's probably never. So I'd be ok removing it. But the consistency with browser history is nice. Then again, we don't support basename in memory history where someone could theoretically do the same thing. So there's an inconsistency somewhere..

**Comment by mjackson** [comment]
Fixed in #432
